test/test_rpc.py

```diff
@@ -133,7 +133,12 @@ def test_reinit(self):
         store = dist.FileStore(self.file.name, self.world_size)
         dist.init_process_group(backend="gloo", rank=self.rank,
                                 world_size=self.world_size, store=store)
-        with self.assertRaisesRegex(RuntimeError, "is not unique"):
+        dist.init_model_parallel(self_name='worker{}'.format(self.rank),
+                                 backend=BACKEND,
+                                 self_rank=self.rank,
+                                 init_method=RPC_INIT_URL)
+        with self.assertRaisesRegex(RuntimeError,
+                                    "(is not unique|is already initialized)"):
```

Comment:
Why can it be both?

Reply:
Process group backed implementation throws "is not unique" probably from ProcessGroupAgent::collectNames. I feel "is already initialized" may be a more generic way to telling the user what is the error and other implementations should throw this instead. We can update pg backed to follow the same in a separate diff.